pkg/loki/modules.go

```diff
@@ -308,7 +320,7 @@ func (t *Loki) initQueryFrontend() (_ services.Service, err error) {
 	t.server.HTTP.Handle("/api/prom/label/{name}/values", frontendHandler)
 	t.server.HTTP.Handle("/api/prom/series", frontendHandler)
 	// fallback route
-	t.server.HTTP.PathPrefix("/").Handler(frontendHandler)
+	t.server.HTTP.PathPrefix("/").Handler(tailProxy)
```

Comment:
The default value for `tailproxy.downstream-url` is empty string and not everyone would want to set it but here we use it anyways. I think we should use it only when a value is set for it.

Reply:
oh great point I will fix it

Reply:
great point thanks, I will fix it

Reply:
what do you think about this solution https://github.com/grafana/loki/pull/2032/commits/ef61f933f0a4b764a2ee7cfdc3487ce9cc8feed8?